include/nbl/builtin/hlsl/bxdf/ndf/beckmann.hlsl

```diff
@@ -338,18 +322,15 @@ struct Beckmann
         return retval;
     }
 
-    template<class MicrofacetCache, typename C=bool_constant<!IsBSDF> NBL_FUNC_REQUIRES(RequiredMicrofacetCache<MicrofacetCache>)
-    enable_if_t<C::value && !IsBSDF, quant_query_type> createQuantQuery(NBL_CONST_REF_ARG(MicrofacetCache) cache, scalar_type orientedEta)
+    template<class MicrofacetCache NBL_FUNC_REQUIRES(RequiredMicrofacetCache<MicrofacetCache>)
+    quant_query_type createQuantQuery(NBL_CONST_REF_ARG(MicrofacetCache) cache, scalar_type orientedEta)
     {
-        quant_query_type dummy; // brdfs don't make use of this
-        return dummy;
-    }
-    template<class MicrofacetCache, typename C=bool_constant<IsBSDF> NBL_FUNC_REQUIRES(RequiredMicrofacetCache<MicrofacetCache>)
-    enable_if_t<C::value && IsBSDF, quant_query_type> createQuantQuery(NBL_CONST_REF_ARG(MicrofacetCache) cache, scalar_type orientedEta)
-    {
-        quant_query_type quant_query;
-        quant_query.VdotHLdotH = cache.getVdotHLdotH();
-        quant_query.VdotH_etaLdotH = cache.getVdotH() + orientedEta * cache.getLdotH();
+        quant_query_type quant_query;   // only has members for refraction
+        if (SupportsTransmission)
+        {
+            quant_query.VdotHLdotH = cache.getVdotHLdotH();
```

Comment:
add a comment that even though in PBRT its `abs(VdotH)*abs(LdotH)` we leverage that under transmission the sign must always be negattive and rest of the code accounts for that.